src/libasr/codegen/asr_to_c.cpp

```diff
@@ -282,7 +291,34 @@ class ASRToCVisitor : public BaseCCPPVisitor<ASRToCVisitor>
                         use_ref = false;
                         dims = "";
                     }
-                    sub = format_type_c(dims, "struct " + der_type_name + "*",
+                    std::string ptr_char = "*";
+                    if( !use_ptr_for_derived_type ) {
+                        ptr_char.clear();
+                    }
+                    sub = format_type_c(dims, "struct " + der_type_name + ptr_char,
+                                        v.m_name, use_ref, dummy);
+                }
+            } else if (ASR::is_a<ASR::Union_t>(*v.m_type)) {
+                std::string indent(indentation_level*indentation_spaces, ' ');
+                ASR::Union_t *t = ASR::down_cast<ASR::Union_t>(v.m_type);
+                std::string der_type_name = ASRUtils::symbol_name(t->m_union_type);
+                 if( is_array ) {
```

Comment:
```suggestion
                if( is_array ) {
```

Reply:
@czgdp1807 I think you forgot this one.

Reply:
Ah! yes.

Reply:
I will resolve this tomorrow Morning, clean up the git history and merge.

Reply:
Good to merge, this is just a tiny thing anyway.